docs/apache-airflow/security/secrets/index.rst

```diff
@@ -30,6 +30,66 @@ The following are particularly protected:
 .. toctree::
     :maxdepth: 1
     :glob:
+    :caption: Further reading:
 
-    fernet
-    secrets-backend/index
+    Encryption at rest <fernet>
+    Using external Secret stores <secrets-backend/index>
+
+.. _security:mask-sensitive-values:
+
+Masking sensitive data
+----------------------
+
+Airflow will by default mask Connection passwords and sensitive Variables and keys from a Connection's
+extra (JSON) field when they appear in Task logs, or in the Variable view of the UI.
+
+It does this by looking for the specific *value* appearing anywhere in your output. This means that if you
+have a connection with a password of ``a``, then every instance of the letter a in your logs will be repleaced
+with ``***``.
+
+To disable masking you can setting :ref:`config:core__hide_sensitive_var_conn_fields` to False.
+
+Sensitive field names
+"""""""""""""""""""""
+
+When masking is enabled, Airflow will always mask the password field of every Connection that is accessed by a
+task.
+
+It will also mask the value of a Variable, or the field of a Connection's extra JSON blob if the name contains
+any words in ('password', 'secret', 'passwd', 'authorization', 'api_key', 'apikey', 'access_token'). This list
+can also be extended:
+
+.. code-block:: ini
+
+    [core]
+    sensitive_var_conn_names = comma,separated,sensitive,names
+
+Adding your own masks
+"""""""""""""""""""""
+
+If you want to mask an additional secret that is already masked by one of the above methods, you can do it in
+your DAG file or operator's ``execute`` function using the ``mask_secret`` function. For example:
+
+.. code-block:: python
+
+    @task
+    def my_func():
+        from airflow.utils.log.secrets_masker import mask_secret
```

Comment:
I'm not 100% sold that this is the _right_ place for users to import from.

I'm half tempted to make this a lazy import in `airflow/__init__.py` so this becomes

```suggestion
        from airflow import mask_secret
```

Anyone have any thoughts?

Reply:
This doesn’t feel like a top level import to me. Maybe the import path should be shorter, but probably not *that* short.

Reply:
I couldn't think of any ideas for what a shorter import path might look like without being top level 😁

Reply:
I guess it could be `airflow.utils.mask_secret`

Reply:
Or `airflow.log.mask_secret`? (Is it generally frawned upon to add a new module under `airflow`?)

Reply:
Nope, though as soon as we do that _I'm_ going to want to move a whole load of modules out of airflow.utils 😁  (probably no bad thing)

Reply:
I’m not going to complain if you do.

Reply:
Tell you what -- I'll leave this here for this PR, then follow up with a refactor.

Reply:
Naming: `airflow.log` vs `airflow.logs` vs `airflow.logging`

Any preference @uranusjr ?